### PR TITLE
Users can create, view, and delete categories

### PR DESCRIPTION
**Pull request by hrs**
* Create a Category model, which just has a (unique) name.
* Display an alphabetized list of all the categories.
* View existing categories.
* Create new categories (from a form on the index page).
* Delete categories.

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -14,10 +14,11 @@ module Features
   include Formulaic::Dsl
 end
 
+Capybara::Webkit.configure do |config|
```
Comment (hrs) [inline spec/rails_helper.rb]: I think that's a bad idea for a config block.

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  def show
+    @category = find_category
+  end
+
+  def create
+    @category = Category.new(category_params)
+
+    if !@category.save
+      flash[:error] = "#{@category.errors.full_messages.join(", ")}! D:"
```
Comment (hrs) [inline app/controllers/categories_controller.rb]: Nope, it reads fine this way.

File: app/controllers/categories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def show
+    @category = find_category
+  end
+
+  def create
+    @category = Category.new(category_params)
+
+    if !@category.save
```
Comment (hrs) [inline app/controllers/categories_controller.rb]: I refuse to favor `unless`.

File: app/controllers/categories_controller.rb (hunk 2 above)
Comment (jencheng) [inline app/controllers/categories_controller.rb]: hahahaha "D:"

File: spec/factories.rb [hunk 4]
```diff
@@ -1,4 +1,8 @@
 FactoryGirl.define do
+  factory :category do
+    sequence(:name) { |n| "category_#{n}" }
+  end
+
   factory :user do
```
Comment (jencheng) [inline spec/factories.rb]: tests :dancer:

**Comment by jencheng** [comment]
looks good!

**Pull request merged**